src/test-utils.ts

```diff
@@ -0,0 +1,3 @@
+export const asMockedFn = <T extends (...args: any[]) => any>(fn: T): jest.MockedFunction<T> => {
```

Comment:
I would add a comment here:
// type "any" is used here to allow for desired type flow during usage where T "looks like a function"

also, a jsdoc comment
/*
 * Use when working with a jest.fn() mocked method to get better type safety and IDE hinting on 
 * the function signature of what's being mocked
 */

Reply:
Once #3476 is merged, this can also use `AnyFn`.